xarray/backends/cfgrib_.py

```diff
@@ -90,7 +90,8 @@ def get_encoding(self):
         return {"unlimited_dims": {k for k, v in dims.items() if v is None}}
 
 
-class CfgribfBackendEntrypoint(BackendEntrypoint):
+class CfgribfBackendEntrypoint(_InternalBackendEntrypoint):
```

Comment:
Is it worrisome that xarray warmly recommends 3rd party backend creators to use BackendEntrypoint but xarray, whos backends will be used as reference, doesn't use it?

Maybe _InternalBackendEntrypoint can be viewed as a experimental version and the good ideas will eventually end up in BackendEntrypoint as well?

Reply:
The "problem" was that internal backends need the `available` attribute while external backends do not need it. When xarray finds them in the entry points they will be available.

But maybe I can refactor the logic of available backends and remove this logic from the BackEntryPoint completely and move it elsewhere.

Reply:
Can we add internal backends to the entry points only if they are available?

Reply:
Usually the entry points are defined in the `setup.py` and therefore set during installation of a package.

Theoretically it is possible to dynamically add entries but I feel that this would make things only more complicated.

Reply:
I think some slight refactoring can solve this issue:

We can simply move the logic that checks whether engines are available to `build_engines`.
Then we don't need to have the internal backend class anymore.

Reply:
Check if you prefer the new implementation :)